trading_calendars/calendar_utils.py

```diff
@@ -162,15 +162,18 @@ def __init__(self, calendars, calendar_factories, aliases):
         self._calendar_factories = dict(calendar_factories)
         self._aliases = dict(aliases)
 
-    def get_calendar(self, name):
+    def get_calendar(self, name, **kwargs):
         """
         Retrieves an instance of an TradingCalendar whose name is given.
 
         Parameters
         ----------
         name : str
             The name of the TradingCalendar to be retrieved.
-
+        kwargs: Dict[str, Any]
+            Optional keyword args passed to calendar `__init__.
+            Note: Arguments are only passed when the calendar is first
+            constructed. Subsequent calls will not affect the calendar.
```

Comment:
I can see the argument for moving `start_default`/`end_default` to the dispatcher and allowing you to change that globally (as properties on the dispatcher instance?), but I think the interface in this PR targets the most common use case (though I could be mistaken) when you're grabbing a calendar out of a registry. I wanted to keep it very straightforward to pull out preexisting calendars. I don't want to force users to go re-register a bunch of calendars. 

I can definitely add a warning here if you don't think that would be too noisy. 

> Looking at that original zipline issue

I think zipline should either pass the dates it wants here (i.e. passing the `start` in https://github.com/quantopian/zipline/blob/master/zipline/__main__.py#L290) or we should clip the data bundles.

Reply:
> I think zipline should either pass the dates it wants here (i.e. passing the start in https://github.com/quantopian/zipline/blob/master/zipline/__main__.py#L290) or we should clip the data bundles.

If the expectation is that that particular call site will actually construct the calendar, that feels brittle to me, since it's not enforced. Every third-party extension or zipline module will have to be sure not to call `get_calendar` first. For this case, have you confirmed that that is the first call site to run (at least on master currently without extensions)? We could solve that potentially with a cli unit test, but my preference is to have more robust underpinnings, since I'd hate to cause more confusion when writing an extension, i.e. if you call `get_calendar` too early, it will mess zipline up.

Reply:
> If the expectation is that that particular call site will actually construct the calendar, that feels brittle to me, since it's not enforced. 

Then should we have zipline not use the registry? I can add some sort of `no_cache` option to `get_calendar` as well so zipline can forcibly recreate the calendar it needs to, but I'm not sure using the global dispatcher is right then. 

In general, I feel that we should lean towards making the standalone usage of `trading_calendars` more ergonomic and making changes in zipline we need to in order to support that (though this PR would keep zipline working as is). 

>  For this case, have you confirmed that that is the first call site to run (at least on master currently without extensions)?

If you use the cli main entry point, that's the first instance we grab the calendar.